src/nwb_datajoint/common/common_metrics.py

```diff
@@ -0,0 +1,153 @@
+        self.insert1([list_name, params], skip_duplicates=True)
+    
+    # TODO
+    def _validate_metrics_list(self, key):
+        """Checks whether a row to be inserted contains only the available metrics
+        """
+        # get available metrics list
+        # get metric list from key
+        # compare
+        return NotImplementedError
+    
+@schema
+class MetricSelection(dj.Manual):
+    definition="""
+    -> Waveforms
+    -> MetricParameters
+    ---
+    """        
+    
+@schema
+class QualityMetrics(dj.Computed):
+    definition="""
+    -> MetricSelection    
+    ---
+    quality_metrics_path: varchar(500)
+    -> AnalysisNwbfile
+    object_id: varchar(40) # Object ID for the metrics in NWB file
+    """
+    def make(self, key):
+        waveform_extractor = Waveforms().load_waveforms(key)
+        qm = {}
+        params = (MetricParameters & key).fetch1('metric_params')
+        for metric_name, metric_params in params.items():
+            metric = self._compute_metric(waveform_extractor, metric_name, **metric_params)
+            qm[metric_name] = metric
+        qm_name = self._get_quality_metrics_name(key)
+        key['quality_metrics_path'] = str(Path(os.environ['NWB_DATAJOINT_WAVEFORMS_DIR']) / Path(qm_name+'.json'))
+        # save metrics dict as json
+        print(f'Computed all metrics: {qm}')
+        self._dump_to_json(qm, key['quality_metrics_path'])
+
+        key['analysis_file_name'] = AnalysisNwbfile().create(key['nwb_file_name'])
+        key['object_id'] = AnalysisNwbfile().add_units_metrics(key['analysis_file_name'], metrics=qm)
+        AnalysisNwbfile().add(key['nwb_file_name'], key['analysis_file_name'])       
+        
+        self.insert1(key)
+        
+    def _get_quality_metrics_name(self, key):
+        wf_name = Waveforms()._get_waveform_extractor_name(key)
+        qm_name = wf_name + '_qm'
+        return qm_name
+    
+    def _compute_metric(self, waveform_extractor, metric_name, **metric_params):
+        metric_func = _metric_name_to_func[metric_name]
+        if metric_name == 'snr' or metric_name == 'isi_violation':
+            metric = metric_func(waveform_extractor, **metric_params)
+        else:
+            metric = {}
+            for unit_id in waveform_extractor.sorting.get_unit_ids():
+                metric[str(unit_id)] = metric_func(waveform_extractor, this_unit_id=unit_id, **metric_params)
+        return metric
+    
+    def _dump_to_json(self, qm_dict, save_path):
+        new_qm = {}
+        for key, value in qm_dict.items():
+            m = {}
+            for unit_id, metric_val in value.items():
+                m[str(unit_id)] = metric_val
+            new_qm[str(key)] = m
+        with open(save_path, 'w', encoding='utf-8') as f:
+            json.dump(new_qm, f, ensure_ascii=False, indent=4)
+        
+def _get_metric_default_params(metric):
+    if metric == 'nn_isolation':
```

Comment:
This is probably more straightforward as a dictionary of dictionaries rather than a bunch of if else statements.

Reply:
converted to dict of dict